### PR TITLE
Message signing

### PR DESCRIPTION
**Pull request by janmazak**
# Checklist

- [ ] App update process has been followed 
- [ ] Target branch is `develop` 
- [ ] Application version has been bumped

### CODE REVIEW

**Comment by janmazak** [comment]
Hash should be displayed always (even if msg is not hashed), I'll have to do it yet.

**Comment by janmazak** [comment]
Hash displayed now.

The fuzzer does not work yet, it needs a new harness for the new signMsg call.

File: src/signMsg_ui.c [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+			uint8_t witnessKey[PUBLIC_KEY_SIZE];
+		} wireResponse = {0};
+
+		STATIC_ASSERT(SIZEOF(ctx->signature) == ED25519_SIGNATURE_LENGTH, "wrong signature buffer size");
+		memmove(wireResponse.signature, ctx->signature, ED25519_SIGNATURE_LENGTH);
+
+		STATIC_ASSERT(SIZEOF(ctx->witnessKey) == PUBLIC_KEY_SIZE, "wrong key buffer size");
+		memmove(wireResponse.witnessKey, ctx->witnessKey, PUBLIC_KEY_SIZE);
+
+		io_send_buf(SUCCESS, (uint8_t*) &wireResponse, SIZEOF(wireResponse));
```
Comment (davidmisiak) [inline src/signMsg_ui.c:315]: The value of the address field should be returned.

Comment (davidmisiak) [reply src/signMsg_ui.c:315]: (already mentioned in ledgerjs PR comments)

**Review by davidmisiak — COMMENTED** [review]
Let me know if I should do some manual tests.

**Comment by janmazak** [comment]
Signatures passed verification by M. Lang.

File: src/signMsg.c [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+			if (ctx->receivedChunks == 2) {
+				if (ctx->isAscii) {
+					VALIDATE(previousSize == MAX_CIP8_MSG_FIRST_CHUNK_ASCII_SIZE, ERR_INVALID_DATA);
+				} else {
+					VALIDATE(previousSize == MAX_CIP8_MSG_FIRST_CHUNK_HEX_SIZE, ERR_INVALID_DATA);
+				}
+			} else if (ctx->receivedChunks > 2) {
+				VALIDATE(previousSize == MAX_CIP8_MSG_HIDDEN_CHUNK_SIZE, ERR_INVALID_DATA);
+			}
+		}
```
Comment (davidmisiak) [inline src/signMsg.c:116]: Why do we need to perform this check for the previous chunk when the same conditions are checked below for the current chunk?

**Review by davidmisiak — APPROVED** [review]
I spotted no more issues :+1:

**Pull request merged**